app/src/main/java/com/limelight/binding/input/virtual_controller/VirtualMouse.java

```diff
@@ -0,0 +1,145 @@
+/**
+ * Created by Karim Mreisi.
+ */
+
+package com.limelight.binding.input.virtual_controller;
+
+import android.content.Context;
+import android.os.SystemClock;
+import android.view.MotionEvent;
+import android.view.View;
+import android.view.inputmethod.InputMethodManager;
+
+import com.limelight.binding.input.touch.TouchContext;
+
+/**
+ * This is a copy of the full-screen touchpad that lives under a VirtualController
+ *
+ */
+public class VirtualMouse extends View {
```

Comment:
Can you figure out a way to avoid having to copy this code? It will create extra maintenance work.

Maybe you can pass the StreamView into VirtualController and it can pass along the touch events?

Reply:
Great suggestion! The context object in VirtualMouse is the Game object that contains the mouse code, so the change was very simple. Please check if this is a sensible approach.

Reply:
It's still a bit more code change than I was hoping for. I wonder if there's a way to use the RelativeLayout to pass touch events to StreamView. Perhaps either having the StreamView be a child of VirtualController's RelativeLayout or implementing `onTouchEvent()` to pass the event to StreamView if nothing in the RelativeLayout hierarchy matches.

Reply:
Simply adding streamView as a child of VirtualController's RelativeLayout seems to work fine!

Reply:
This causes a crash:
```
     Caused by: java.lang.IllegalStateException: The specified child already has a parent. You must call removeView() on the child's parent first.
        at android.view.ViewGroup.addViewInner(ViewGroup.java:5247)
        at android.view.ViewGroup.addView(ViewGroup.java:5076)
        at android.view.ViewGroup.addView(ViewGroup.java:5016)
        at android.view.ViewGroup.addView(ViewGroup.java:4988)
        at com.limelight.binding.input.virtual_controller.VirtualController.refreshLayout(VirtualController.java:168)
        at com.limelight.Game.onCreate(Game.java:448)
        at android.app.Activity.performCreate(Activity.java:8256)
        at android.app.Activity.performCreate(Activity.java:8236)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1339)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3660)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3816) 
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:99) 
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135) 
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2299) 
        at android.os.Handler.dispatchMessage(Handler.java:106) 
        at android.os.Looper.loopOnce(Looper.java:201) 
        at android.os.Looper.loop(Looper.java:288) 
        at android.app.ActivityThread.main(ActivityThread.java:7844) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936) 
```